src/metadata/EntityMetadata.ts

```diff
@@ -673,12 +687,18 @@ export class EntityMetadata {
         secondEntity: ObjectLiteral,
     ): boolean {
         const firstEntityIdMap = this.getEntityIdMap(firstEntity)
-        if (!firstEntityIdMap) return false
+        const firstEntityIdFlat = this.getEntityIdFlat(firstEntity)
```

Comment:
When the function is called while inserting multiple objects into the database without specifying the full PK (the database generates it), it returns `true` and only a single object will be inserted, I'd suggest we add a bit of logic to make sure every part of a composite PK is set, WDYT?
```suggestion
        for (const c of this.primaryColumns) {
            if (!Object.hasOwn(firstEntity, c.databaseName) || 
                !Object.hasOwn(secondEntity, c.databaseName)) {
                    return false
            }
        }
        const firstEntityIdFlat = this.getEntityIdFlat(firstEntity)
```

Reply:
Yeah! I missed that case! I like your suggestion! Thanks @andreamouraud! 💪🏼

Reply:
@andreamouraud can you provide a test case to confirm that a new adjustment complies with "inserting multiple objects into the database without specifying the full PK (the database generates it)" 🙂